core/controllers/admin.py

```diff
@@ -854,3 +854,37 @@ def post(self):
             self.render_json({})
         else:
             raise self.InvalidInputException('This app cannot send emails.')
+
+
+class UpdateUsernameHandler(base.BaseHandler):
+    """Handler for renaming usernames."""
+
+    @acl_decorators.can_access_admin_page
+    def put(self):
+        current_username = self.payload.get('current_username', None)
+        new_username = self.payload.get('new_username', None)
+
+        if new_username is None or current_username is None:
+            raise self.InvalidInputException(
+                'Please ensure that the usernames are not empty.')
+
+        if type(current_username).__name__ != 'unicode' or type(
+                new_username).__name__ != 'unicode':
+            raise self.InvalidInputException(
+                'Please ensure that the usernames are strings.')
+
+        user_id = user_services.get_user_id_from_username(current_username)
+        if user_id is None:
+            raise self.InvalidInputException('User does not exist.')
+
+        if len(new_username) > constants.MAX_USERNAME_LENGTH:
+            raise self.InvalidInputException(
+                'Please make sure that the new username is not longer '
```

Comment:
`Expected new_username not to be more than %s characters, received %s`

Reply:
Done, thanks!